src/server/components/dashboard/listsItems/__tests__/requestValidation.test.ts

```diff
@@ -0,0 +1,54 @@
+import { requestValidation } from "../requestValidation";
+import { HttpException } from "../../../../middlewares/error-handlers";
+
+const next = jest.fn();
+let req, res;
+
+beforeEach(() => {
```

Comment:
@RichardBray / @alisalaman-cyb 

FWIW - some slightly nicer testing. A lot of the controllers.ts files have _a lot_ of controllers in them. There should be up to 4 in one file, tbh. One per CRUD operation.  Ideally 1 controller per file. Then it makes writing tests a lot nicer. 

Also, with jest, you do not have to wrap in `describe` especially if you are testing 1 thing per file. 

Each test file is already insulated. `beforeEach` here runs before each top level `test("..")`, so wrapping in `describe` is redundant. jest already labels which file it is (i.e. requestValidation.test.ts).

Reply:
Most OO languages will want you to have 1 thing per file (although sometimes it can get quite "bitty", like for helper/util files which might just be a couple lines each.)

4 controllers in one file would be (let's say we have delete and put route in here too..) 


```
listRouter.get("/:listId/items/:listItemId", controllers.listItemGetController);
listRouter.post("/:listId/items/:listItemId", controllers.listItemPostController);
listRouter.put("/:listId/items/:listItemId", controllers.listItemPutController);
listRouter.delete("/:listId/items/:listItemId", controllers.listItemDeleteController);
```

so you can refactor to

```.ts
// list/listItem/controllers.ts (or list/listItem/controller/index.ts)
export const { getController };
export const { postController };
export const { putController };
export const { deleteController };

// if they are each very big controllers, you can split them into their own files.
// e.g. export { getController } from './get'

// to reduce context duplication further you can do 
// export const { get, post, put, delete }; .. but actually I think delete is a reserved word, so you need to call it del 🙂. 

export { get } from './get';
export { post } from './post';
export { del } from './del';
export { put } from './put';

```

```
import * as listItemControllers from 'list/listItem/controllers';

listRouter.get("/:listId/items/:listItemId", listItemControllers.get);
listRouter.post("/:listId/items/:listItemId", listItemControllers.post);
listRouter.put("/:listId/items/:listItemId", listItemControllers.put);
listRouter.delete("/:listId/items/:listItemId", listItemControllers.del);
```

If you think, `listItemController` as a single class, the only exposed (public) methods will be each of the CRUD operations, based on our usage of controllers (route handlers). If there is code shared between the controllers, then it's a helper/util. Put them in a sibling file and import them. 

I don't think it's documented anywhere like that exactly, but it implements SRP/SOLID principles. Svelte(kit) also implements this.
`blog/page.svelte` <- get request (by default), route is /blog 
`blog/+server.js`  <- you can put POST / PUT / PATCH / DELETE functions in here. if you PUT to /blog, it will call the put function. same with POST, DELETE etc.

Reply:
Yep, so do I. Directory based routing FTW. Thanks for the explanation 🙇🏿